workspaces/api-gateway/src/__tests__/api/auth/SignInService.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { genSalt, hash } from 'bcrypt';
+import { DataSource } from 'typeorm';
+import { JwtService } from '@nestjs/jwt';
+import type { Repository } from 'typeorm';
+
+import { User } from 'database/models/user.entity';
+import { AuthModule } from 'api/auth/auth.module';
+import { createTestingModule } from '__tests__/utils';
+import { userData } from '__tests__/seeds/user';
+import { SignInService } from 'api/auth/services/SignInService';
+import type { AuthCredentialsDto } from 'api/auth/dto/auth-credential.dto';
+
+describe('SignInService', () => {
+  let userRepository: Repository<User>;
+  let dataSource: DataSource;
+  let jwtService: JwtService;
+  let signInService: SignInService;
+  let user: User;
+
+  const userDto: AuthCredentialsDto = {
+    username: userData.username,
+    password: userData.password,
+  };
+
+  beforeAll(async () => {
+    const { module } = await createTestingModule({
+      imports: [AuthModule],
+      entities: [User],
+    });
+
+    userRepository = module.get('UserRepository');
+    dataSource = module.get(DataSource);
+    signInService = module.get(SignInService);
+    jwtService = module.get(JwtService);
+    jest.spyOn(jwtService, 'sign').mockImplementation(() => 'fakeToken');
+  });
+
+  beforeEach(async () => {
+    const salt = await genSalt();
+    const hashedPassword = await hash(userDto.password, salt);
+    user = userRepository.create({
+      username: userDto.username,
+      password: hashedPassword,
+    });
+    await userRepository.save(user);
+  });
```

Comment:
every data creation must be in `beforeAll` and every data clear must be in `afterAll`